Trilinos nightly failure, ifpack2: spiluk errors with too few arguments to function call

Trilinos nightly builds are failing to compile in ifpack2 (all backends) with the following errors:

```
17:51:07 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/ifpack2/src/Ifpack2_IlukGraph.hpp:598:98: error: too few arguments to function call, expected 4, have 3
17:51:07                                                     KernelHandle->get_spiluk_handle()->get_nnzU());
17:51:07                                                                                                  ^
17:51:07 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/ifpack2/src/Ifpack2_RILUK_def.hpp:620:17: note: in instantiation of member function 'Ifpack2::IlukGraph<Tpetra::CrsGraph<>, KokkosKernels::Experimental::KokkosKernelsHandle<const unsigned long, const int, double, Kokkos::Serial, Kokkos::HostSpace, Kokkos::HostSpace>>::initialize' requested here
17:51:07         Graph_->initialize (KernelHandle_); // this calls spiluk_symbolic
17:51:07                 ^
17:51:07 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/kokkos-kernels/sparse/src/KokkosSparse_spiluk_handle.hpp:134:8: note: 'reset_handle' declared here
17:51:07   void reset_handle(const size_type nrows_, const size_type nnzL_,
17:51:07        ^
17:51:07 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Build/packages/ifpack2/src/Ifpack2_RILUK_DOUBLE_INT_LONG_LONG_SERIAL.cpp:49:
17:51:07 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/ifpack2/src/Ifpack2_RILUK_decl.hpp:55:
17:51:07 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/ifpack2/src/Ifpack2_IlukGraph.hpp:618:76: error: too few arguments to function call, expected 4, have 3
17:51:07       KernelHandle->get_spiluk_handle()->reset_handle(NumMyRows, nnzL, nnzU);
17:51:07       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                      ^
17:51:07 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/kokkos-kernels/sparse/src/KokkosSparse_spiluk_handle.hpp:134:8: note: 'reset_handle' declared here
17:51:07   void reset_handle(const size_type nrows_, const size_type nnzL_,
```

This follows from merge of #2064 , @jgfouca can you take a look? It's not clear to me if the changes introduced a compatibility-breaking change which will need to be addressed and guarded (by version) in Trilinos, or perhaps a missing overload to handle function calls with fewer args (I'm guessing exec space instance maybe)?

**Reproducer** (Blake, all queue):
```bash
# Repos
git clone -b develop https://github.com/trilinos/Trilinos.git
TRILINOS_DIR=$PWD/Trilinos
git clone -b develop https://github.com/kokkos/kokkos.git
KOKKOS_DIR=$PWD/kokkos
git clone -b develop https://github.com/kokkos/kokkos-kernels.git
KOKKOSKERNELS_DIR=$PWD/kokkos-kernels

# Grab a node
salloc -N 1 -p all

# Environment
module load cmake llvm openblas/0.3.23

# Configure
cmake \
  -D CMAKE_INSTALL_PREFIX="${TRILINOS_INSTALL_DIR}" \
  -D CMAKE_CXX_COMPILER="`which clang++`" \
  -D CMAKE_C_COMPILER="`which clang`" \
  -D CMAKE_Fortran_COMPILER="`which gfortran`" \
  -D CMAKE_CXX_FLAGS="-g -fPIC" \
  -D CMAKE_C_FLAGS="-g -fPIC" \
  -D CMAKE_Fortran_FLAGS="-g -fPIC" \
  -D CMAKE_BUILD_TYPE:STRING=DEBUG \
  -D BUILD_SHARED_LIBS:BOOL=OFF \
  -DTPL_ENABLE_MPI=OFF \
  -DTPL_ENABLE_BLAS:BOOL=ON \
   -DBLAS_LIBRARY_DIRS:FILEPATH=${OPENBLAS_ROOT}/lib \
   -DBLAS_LIBRARY_NAMES:STRING="openblas" \
  -DTPL_ENABLE_LAPACK:BOOL=ON \
   -DLAPACK_INCLUDE_DIRS:FILEPATH=${OPENBLAS_ROOT}/include \
   -DLAPACK_LIBRARY_DIRS:FILEPATH=${OPENBLAS_ROOT}/lib \
   -DLAPACK_LIBRARY_NAMES:STRING="openblas" \
  -DTrilinos_ENABLE_ALL_PACKAGES=OFF \
  -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF \
  -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_MUST_FIND_ALL_TPL_LIBS=TRUE \
  -DTrilinos_ENABLE_COMPLEX=ON \
  -DTrilinos_ENABLE_OpenMP=OFF \
  -DTrilinos_ENABLE_Kokkos=ON \
  -D Kokkos_ENABLE_SERIAL=ON \
   -D Kokkos_ENABLE_TESTS=ON \
  -D Kokkos_ARCH_SKX=ON \
  -DTrilinos_ENABLE_KokkosKernels=ON \
   -D KokkosKernels_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Tpetra=ON \
   -D Tpetra_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Sacado=ON \
   -D Sacado_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Stokhos=ON \
   -D Stokhos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Intrepid2=ON \
   -D Intrepid2_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Adelus=ON \
   -D Adelus_ENABLE_TESTS:BOOL=ON \
  -DTrilinos_ENABLE_Compadre=ON \
   -D Compadre_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Panzer=ON \
   -D Panzer_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Amesos2=ON \
   -D Amesos2_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Zoltan2=ON \
   -D Zoltan2_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Teuchos=ON \
   -D Teuchos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Ifpack2=ON \
   -D Ifpack2_ENABLE_TESTS=ON \
\
  -DTPL_ENABLE_Matio=OFF \
\
-DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
-DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
\
$TRILINOS_DIR

# Compile
make -j16
```

Most of the packages enabled can be removed, probably only need to enable ifpack2 and its tests (snip above is from the nightly job)

--- a/sparse/src/KokkosSparse_spiluk_handle.hpp
+++ b/sparse/src/KokkosSparse_spiluk_handle.hpp
@@ -132,21 +132,26 @@ class SPILUKHandle {
         vector_size(-1) {}
 
   void reset_handle(const size_type nrows_, const size_type nnzL_,
-                    const size_type nnzU_, const size_type block_size_) {
+                    const size_type nnzU_, const size_type block_size_ = -1) {
     set_nrows(nrows_);
     set_num_levels(0);
     set_nnzL(nnzL_);
     set_nnzU(nnzU_);
-    set_block_size(block_size_);
+    // user likely does not want to reset block size to 0, so set default
+    // to -1
+    if (block_size_ != static_cast<size_type>(-1)) {
+      set_block_size(block_size_);
+    }
     set_level_maxrows(0);
     set_level_maxrowsperchunk(0);
-    level_list          = nnz_row_view_t("level_list", nrows_),
-    level_idx           = nnz_lno_view_t("level_idx", nrows_),
-    level_ptr           = nnz_lno_view_t("level_ptr", nrows_ + 1),
-    hlevel_ptr          = nnz_lno_view_host_t("hlevel_ptr", nrows_ + 1),
-    level_nchunks       = nnz_lno_view_host_t(),
-    level_nrowsperchunk = nnz_lno_view_host_t(), reset_symbolic_complete(),
+    level_list          = nnz_row_view_t("level_list", nrows_);
+    level_idx           = nnz_lno_view_t("level_idx", nrows_);
+    level_ptr           = nnz_lno_view_t("level_ptr", nrows_ + 1);
+    hlevel_ptr          = nnz_lno_view_host_t("hlevel_ptr", nrows_ + 1);
+    level_nchunks       = nnz_lno_view_host_t();
+    level_nrowsperchunk = nnz_lno_view_host_t();
     iw                  = work_view_t();
+    reset_symbolic_complete();
   }
 
   virtual ~SPILUKHandle(){};

--- a/sparse/src/KokkosSparse_spiluk_handle.hpp
+++ b/sparse/src/KokkosSparse_spiluk_handle.hpp
@@ -131,8 +131,9 @@ class SPILUKHandle {
         team_size(-1),
         vector_size(-1) {}
 
-  void reset_handle(const size_type nrows_, const size_type nnzL_,
-                    const size_type nnzU_, const size_type block_size_ = Kokkos::ArithTraits<size_type>::max()) {
+  void reset_handle(
+      const size_type nrows_, const size_type nnzL_, const size_type nnzU_,
+      const size_type block_size_ = Kokkos::ArithTraits<size_type>::max()) {
     set_nrows(nrows_);
     set_num_levels(0);
     set_nnzL(nnzL_);

--- a/sparse/src/KokkosSparse_spiluk_handle.hpp
+++ b/sparse/src/KokkosSparse_spiluk_handle.hpp
@@ -132,14 +132,14 @@ class SPILUKHandle {
         vector_size(-1) {}
 
   void reset_handle(const size_type nrows_, const size_type nnzL_,
-                    const size_type nnzU_, const size_type block_size_ = -1) {
+                    const size_type nnzU_, const size_type block_size_ = Kokkos::ArithTraits<size_type>::max()) {
     set_nrows(nrows_);
     set_num_levels(0);
     set_nnzL(nnzL_);
     set_nnzU(nnzU_);
     // user likely does not want to reset block size to 0, so set default
     // to -1
-    if (block_size_ != static_cast<size_type>(-1)) {
+    if (block_size_ != Kokkos::ArithTraits<size_type>::max()) {
       set_block_size(block_size_);
     }
     set_level_maxrows(0);

--- a/sparse/src/KokkosSparse_spiluk_handle.hpp
+++ b/sparse/src/KokkosSparse_spiluk_handle.hpp
@@ -139,7 +139,7 @@ class SPILUKHandle {
     set_nnzL(nnzL_);
     set_nnzU(nnzU_);
     // user likely does not want to reset block size to 0, so set default
-    // to -1
+    // to size_type::max
     if (block_size_ != Kokkos::ArithTraits<size_type>::max()) {
       set_block_size(block_size_);
     }

